pkg/ingestor/api/api.go

```diff
@@ -61,6 +64,11 @@ func (g *IngestorAPI) Ingest(_ context.Context, clusterName string, runID string
 	var err error
 	defer func() { spanJob.Finish(tracer.WithError(err)) }()
 
+	err = g.checkPreviousRun(runCtx, clusterName, runID) //nolint: contextcheck
```

Comment:
small thing but:

I believe the `checkPreviousRun` is confusing a bit:
- what does it checks?
- it returns an error for what?

Imo, this should be named something like "checkAlreadyIngested" and potentially, return `(bool, error)` instead of just `error`, (potentially `isAlreadyIngested`, if you use that bool).
I don't have a strong opinion on the `(bool, error)` part, because semantically, the function is expected to return an error. But in any case, the error should be exported and checked directly against, not simply by `err != nil`. (if there's a blip in the db call or something, it will return an error, while there wasn't a previous ingestion.


The caller (here), could do:
```golang
ErrAlreadyIngested := errors.New("ingestion already completed")
...
...
err := ...
if error.Is(err, ErrAlreadyIngested) {
    return fmt.Errorf("%w [%s:%s]", ErrAlreadyIngested, clusterName, runID)
}
if err != nil {
    return err
}
```

or, with the (bool, error) return
```golang
ErrAlreadyIngested := errors.New("ingestion already completed")
...
...
alreadyIngested, err := ...
if err != nil {
    return err
}
if alreadyIngested {
    return fmt.Errorf("%w [%s:%s]", ErrAlreadyIngested, clusterName, runID)
}
```

Reply:
changed to bool,err with `isAlreadyIngested()`